tfjs-backend-webgpu/src/argminmax_webgpu.ts

```diff
@@ -87,48 +93,17 @@ export class ArgMinMaxProgram implements WebGPUProgram {
 
       ${sharedMemorySnippet}
 
-      // In order to get a flattened index into the input tensor, we need to
-      // add back the index along the reduced dimension to |outputCoords|.
-      // This function outputs the offset to the first value along
-      // |axis| and the stride to get the next value of the input along |axis|.
-      fn getInputCoordInfo(outputIndex : i32) -> vec2<i32>{
-        let outputCoords = getCoordsFromIndex(outputIndex);
-        var i = ${this.outputShape.length - 1};
-
-        var stride = 1;
-        var inputStride = 1;
-        var offset = 0;
-
-        for (var r = 1; r <= ${this.inputShape.length}; r = r + 1) {
-          let length = ${indexInputShape(`${this.inputShape.length} - r`)};
-          if (${this.inputShape.length} - r == uniforms.axis) {
-            inputStride = stride;
-          } else {
-            offset = offset + ${
-        indexOutputCoords('outputCoords', 'i')} * stride;
-            i = i - 1;
-          }
-          stride = stride * length;
-        }
-
-        return vec2<i32>(offset, inputStride);
-      }
-
-      fn getInputIndex(coordInfo : vec2<i32>, index : i32) -> i32{
-        return coordInfo[0] + coordInfo[1] * index;
-      }
-
       ${getMainHeaderAndGlobalIndexString()}
         let outputIndex = index / i32(workGroupSizeX);
-        let coordInfo = getInputCoordInfo(outputIndex);
-        let Length = ${indexInputShape('uniforms.axis')};
+        let Length = ${getInputShapeLastDim()};
 
         var bestIndex = i32(localId.x);
         var bestValue = uniforms.infinityValue;
 
         for (var k = i32(localId.x); k < Length && outputIndex < uniforms.size;
             k = k + i32(workGroupSizeX)) {
-          let candidate = f32(x[getInputIndex(coordInfo, k)]);
+          let outputCoords = getCoordsFromIndex(outputIndex);
+          let candidate = f32(getX(${splitOutputCoords()} k));
```

Comment:
nit: Put `let outputCoords = getCoordsFromIndex(outputIndex);` out of the `for` loop.